File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        
        int count = 0;

        int countMax = 0;

        for(int i=0; i< nums.size(); i++){

            if(nums[i] == 1){
                count ++;
            }
             else {
                countMax = max(countMax, count);
                count = 0;
             }
        }

        if( count > countMax) {
            return count;
        } else {
            return countMax;
        }
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
         

         int count =0;
         int maxi =0;

         for(int i =0; i<nums.length; i++){

            if(nums[i] == 0){
                maxi = Math.max(count,maxi);
                
                count =0;
                continue;
            }

            count++;
         }
  
         maxi =  Math.max(maxi,count);
         return maxi;
    }
}